src/api/pdc_client_connect.c

```diff
@@ -251,6 +256,108 @@ pdc_client_check_int_ret_cb(const struct hg_cb_info *callback_info)
     FUNC_LEAVE(ret_value);
 }
 
+static void *
+hg_progress_fn(void *foo)
+{
+    hg_return_t   ret;
+    unsigned int  actual_count;
+    hg_context_t *hg_context = (hg_context_t *)foo;
+
+#ifdef TANG_DEBUG
+    char cur_time[64];
+    PDC_get_time_str(cur_time);
+    printf("[%s] enter %s \n", cur_time, __func__);
+#endif
+
+    while (hg_progress_flag_g != 1) {
+        do {
+            /* PDC_get_time_str(cur_time); */
+            /* printf("[%s] before HG_Trigger\n", cur_time); */
+
+            ret = HG_Trigger(hg_context, 0, 1, &actual_count);
+
+            /* PDC_get_time_str(cur_time); */
+            /* printf("[%s] after HG_Trigger\n", cur_time); */
+        } while ((ret == HG_SUCCESS) && actual_count && hg_progress_flag_g != 1);
+
+        /* PDC_get_time_str(cur_time); */
+        /* printf("[%s] before HG_Progress\n", cur_time); */
+
+        if (hg_progress_flag_g != 1)
+            HG_Progress(hg_context, 100);
+
+        usleep(1000);
+        /* PDC_get_time_str(cur_time); */
+        /* printf("[%s] after HG_Progress\n", cur_time); */
+    }
+
+#ifdef TANG_DEBUG
```

Comment:
maybe we should not expose our names as macro name?

Reply:
I just want to make it easier to enable/disable the debug messages next time a similar issue comes up.

Reply:
@zhangwei217245 should we go ahead?

Reply:
not a big deal for me